app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -240,4 +240,14 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
             updateOverlayForCurrentState()
         }
     }
+
+    fun addPinnedSiteToTiles() {
+        val adapter = tileContainer.adapter as HomeTileAdapter
+        adapter.updateAdapterSingleInsertion(initHomeTilesCache())
```

Comment:
Okay, red flag, you're calling `init` again - why?

We made a cache so that we don't have to load it from disk all the time - it's even called "cache".